subgrounds/query.py

```diff
@@ -549,7 +550,7 @@ def map(
                 return map_f(new_selection)
 
             case _:
-                raise Exception(f"map: invalid priority {priority}")
+                raise QueryError(f"map: invalid priority {priority}")
```

Comment:
Not sure this should be a QueryError (priority refers to whether the map function is applied to the inner selections before the current selection or vice-versa, has nothing to do with a GraphQL query per say), `SubgroundsError` or even `TransformError` might be more appropriate here since the function is transforming a `Selection` object

(Unless I am misunderstanding the meaning of QueryError)

Reply:
to be honest, I named it based on the file name. I don't really understand the nature of all these errors and how they are handled.

I'm leaning towards deferring a better error setup when we have more time.